TalkAloud/TalkAloud/AudioViewController.swift

```diff
@@ -9,103 +9,60 @@
     
     @IBOutlet var playAudioButton: UIButton!
     @IBOutlet var recordAudioButton: UIButton!
     
-    
     override func viewDidLoad() {
         super.viewDidLoad()
     }
     
-//    // Move to Audio Engine class
-//    func setupRecorder() {
-//        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
-//        let documentsDirectory = paths[0]
-//
-//        let audioFilename = documentsDirectory.appendingPathComponent(fileName)
-//
-//        let settings = [AVFormatIDKey: Int(kAudioFormatAppleLossless), AVEncoderAudioQualityKey : AVAudioQuality.max.rawValue,
-//        AVEncoderBitRateKey : 320000, AVNumberOfChannelsKey: 2, AVSampleRateKey: 44100.0] as [String: Any]
-//
-//        var error: NSError?
-//
-//        do {
-//            audioRecorder = try AVAudioRecorder(url: audioFilename, settings: settings)
-//        } catch {
-//            audioRecorder = nil
-//        }
-//
-//        if let err = error {
-//            print("AVAudioRecorder error: \(err.localizedDescription)")
-//        } else {
-//            audioRecorder.delegate = self
-//            audioRecorder.prepareToRecord()
-//        }
-//    }
-//    // Put in Audio Engine Class
-//    func preparePlayer() {
-//        do {
-//            audioPlayer = try AVAudioPlayer(contentsOf: getFileURL())
-//            audioPlayer.delegate = self
-//            audioPlayer.volume = 1.0
-//        } catch {
-//            if let err = error as Error? {
-//                print("AVAudioPlayer error: \(err.localizedDescription)")
-//                audioPlayer = nil
-//            }
-//        }
-//    }
-//    // Put in Audio Engine Class
-//    func getFileURL() -> URL {
-//        let fileManager = FileManager.default
-//        let urls = fileManager.urls(for: .documentDirectory, in: .userDomainMask)
-//        let documentDirectory = urls[0] as URL
-//        let soundURL = documentDirectory.appendingPathComponent(fileName)
-//        return soundURL
-//    }
-    
     @IBAction func playAndStopButtonAction(_ sender: UIButton) {
         // Use enum to determine state of audio of being playable
-        if sender.titleLabel?.text == "Play" {
+        if audioEngine.audioState == .none {
+            audioEngine.audioState = .play
+        }
+        
+        if audioEngine.audioState == .play {
             recordAudioButton.isEnabled = false
             sender.setTitle("Stop", for: .normal)
-            // Call Audio Engine .player that will call these methods
             audioEngine.preparePlayer()
             audioEngine.audioPlayer.play()
-        } else {
+            audioEngine.audioState = .stop
+        } else if audioEngine.audioState == .stop {
             // Call Audio Engine .stop that will call these methods
             audioEngine.audioPlayer.stop()
+            recordAudioButton.isEnabled = true
             sender.setTitle("Play", for: .normal)
+            audioEngine.audioState = .none
```

Comment:
Encapsulate the state change in their respective methods.

Reply:
Not quite sure what you mean here but the problem with me setting the `audioState` in something like `preparePlayer` is that that I won't call these methods until the `audioState` is set to `.play` of course I could probably change how this functions with further investigation.

Reply:
Make it by default `stopped` and remove the `none` state.